helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -49,6 +51,7 @@ receivers:
       # System processes metrics, disabled by default
       # process:
 
+  {{- if .Values.autodetect.prometheus }}
```

Comment:
Same here, I think the flag should only control adding a prometheus_simple receiver entry but receiver_creator and k8s_observer should stay.